packages/hop-node/src/gasboost/GasBoostTransaction.ts

```diff
@@ -355,74 +470,60 @@ class GasBoostTransaction extends EventEmitter implements providers.TransactionR
   }
 
   private async boost (item: InflightItem) {
-    const gasPrice = await this.getBumpedGasPrice()
-    const maxGasPrice = parseUnits(this.maxGasPriceGwei.toString(), 9)
-    if (gasPrice.gt(maxGasPrice)) {
+    const gasFeeData = await this.getBumpedGasFeeData()
+    const maxGasPrice = this.parseGwei(this.maxGasPriceGwei)
+    const priorityFeePerGasCap = this.parseGwei(this.priorityFeePerGasCap)
+    const isMaxReached = gasFeeData.gasPrice?.gt(maxGasPrice) || gasFeeData.maxPriorityFeePerGas?.gt(priorityFeePerGasCap)
+    if (isMaxReached) {
       if (!this.maxGasPriceReached) {
-        const warnMsg = `max gas price reached. boostedGasPrice: ${gasPrice}, maxGasPrice: ${this.maxGasPriceGwei}. cannot boost`
+        const warnMsg = `max gas price reached. boostedGasFee: (${this.getGasFeeDataAsString(gasFeeData)}, maxGasFee: (gasPrice: ${this.maxGasPriceGwei}, maxPriorityFeePerGas: ${this.priorityFeePerGasCap}). cannot boost`
         this.notifier.warn(warnMsg)
         this.logger.warn(warnMsg)
-        this.emit(State.MaxGasPriceReached, gasPrice, this.boostIndex)
+        this.emit(State.MaxGasPriceReached, gasFeeData.gasPrice, this.boostIndex)
         this.maxGasPriceReached = true
       }
       return
     }
-    const tx = await this._sendTransaction(gasPrice)
+    const tx = await this._sendTransaction(gasFeeData)
 
     this.gasPrice = tx.gasPrice
+    this.maxFeePerGas = tx.maxFeePerGas
+    this.maxPriorityFeePerGas = tx.maxPriorityFeePerGas
     this.boostIndex++
     this.track(tx)
     this.emit(State.Boosted, tx, this.boostIndex)
   }
 
-  private async _sendTransaction (gasPrice: BigNumber):Promise<providers.TransactionResponse> {
+  private async _sendTransaction (gasFeeData: Partial<GasFeeData>):Promise<providers.TransactionResponse> {
     const maxRetries = 10
     let i = 0
     while (true) {
       i++
       try {
         if (i > 1) {
-          gasPrice = await this.getBumpedGasPrice(this.gasPriceMultiplier * i)
+          gasFeeData = await this.getBumpedGasFeeData(this.gasPriceMultiplier * i)
```

Comment:
IIUC, optimism will not bump here. If that is the case, then the `gasFeeData` will use the same `gasPrice` or `maxPriorityFee`. This, I believe, would cause the node to throw an error (as it expects 1.1x on each retry).

If that is all true, then I think we should somehow not bump at all on Optimism. I _think_ this is alright, as optimism txs should go through very quickly, but I believe the code should reflect that we are not re-boosting optimism so that we don't leave it up to a weird edge case.

I may be misunderstanding though so please LMK!

Reply:
for optimism, the gas price is not bumped (always uses market gas price)

![DeepinScreenshot_select-area_20210914141749](https://user-images.githubusercontent.com/168240/133335482-6ef93ebc-89a4-4c7c-b9d8-5f0f01506582.png)

![DeepinScreenshot_select-area_20210914141732](https://user-images.githubusercontent.com/168240/133335480-03723d5b-10c7-4661-bd78-2578b93e243a.png)

the loop you're looking at is specific to when sending the transaction and it throws an "already known error" because a really low gas price was used and the tx with same nonce is still mempool (so it tries again with a higher pas price). It's an edge case for xdai and polygon but wouldn't expect it to occur on optimism or arbitrum because the transaction is rejected or accepted immediately

![DeepinScreenshot_select-area_20210914142149](https://user-images.githubusercontent.com/168240/133336013-d25cf1a4-eef4-4877-914d-271f32b737bc.png)

Reply:
Ok nice I didn't see the `isAlreadyKnown` -- makes sense and SGTM.